### PR TITLE
Harrison/cohere reranker

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

File: langchain/retrievers/document_compressors/cohere_rerank.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from pydantic import root_validator
+
+from langchain.retrievers.document_compressors.base import BaseDocumentCompressor
+from langchain.schema import Document
+from langchain.utils import get_from_dict_or_env
+
+
+class CohereRerank(BaseDocumentCompressor):
+    client: Any
+    top_n: int = 3
```
Comment (dev2049) [inline langchain/retrievers/document_compressors/cohere_rerank.py:12]: we usually call this k/top_k, dunno if we prefer being consistent internally or w.r.t. the client api

File: langchain/retrievers/document_compressors/cohere_rerank.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+from pydantic import root_validator
+
+from langchain.retrievers.document_compressors.base import BaseDocumentCompressor
+from langchain.schema import Document
+from langchain.utils import get_from_dict_or_env
+
+
+class CohereRerank(BaseDocumentCompressor):
+    client: Any
```
Comment (dev2049) [inline langchain/retrievers/document_compressors/cohere_rerank.py:11]: nit: should we type hint

**Review by dev2049 — APPROVED** [review]
noice

File: langchain/retrievers/document_compressors/cohere_rerank.py (hunk 1 above)
Comment (hwchase17) [reply langchain/retrievers/document_compressors/cohere_rerank.py:12]: gonna pref top_n - if we had a standard reranking class may rename, but until then will keep separate (if we want to change in future, with pydantic alias it is easy)

File: langchain/retrievers/document_compressors/cohere_rerank.py (hunk 2 above)
Comment (hwchase17) [reply langchain/retrievers/document_compressors/cohere_rerank.py:11]: yup nice

**Pull request merged**